Microsoft.Identity.Web/WebAppServiceCollectionExtensions.cs

```diff
@@ -164,82 +114,152 @@ public static IServiceCollection AddMsal(this IServiceCollection services, IConf
+        /// <param name="builder">AuthenticationBuilder to which to add this configuration</param>
+        /// <param name="configSectionName">The configuration section with the necessary settings to initialize authentication options</param>
+        /// <param name="configuration">The IConfiguration object</param>
+        /// <param name="configureOptions">An action to configure OpenIdConnectOptions</param>
+        /// <param name="openIdConnectScheme">The OpenIdConnect scheme name to be used. By default it uses "OpenIdConnect"</param>
+        /// <param name="cookieScheme">The Cookies scheme name to be used. By default it uses "Cookies"</param>
+        /// <param name="subscribeToOpenIdConnectMiddlewareDiagnosticsEvents">
+        /// Set to true if you want to debug, or just understand the OpenIdConnect events.
+        /// </param>
+        /// <returns></returns>
+        public static AuthenticationBuilder AddSignIn(
+            this AuthenticationBuilder builder,
+            string configSectionName,
+            IConfiguration configuration,
+            string openIdConnectScheme,
+            string cookieScheme,
+            Action<OpenIdConnectOptions> configureOptions,
+            bool subscribeToOpenIdConnectMiddlewareDiagnosticsEvents = false)
         {
-            if (options.SameSite == SameSiteMode.None)
+            builder.Services.Configure(openIdConnectScheme, configureOptions);
+            builder.Services.Configure<MicrosoftIdentityOptions>(options => configuration.Bind(configSectionName, options));
+
+            var microsoftIdentityOptions = configuration.GetSection(configSectionName).Get<MicrosoftIdentityOptions>();
+            var b2COidcHandlers = new AzureADB2COpenIDConnectEventHandlers(openIdConnectScheme, microsoftIdentityOptions);
+
+            builder.AddCookie(cookieScheme);
+            builder.AddOpenIdConnect(openIdConnectScheme, options =>
             {
-                var userAgent = httpContext.Request.Headers["User-Agent"].ToString();
-                if (disallowsSameSiteNone(userAgent))
+                options.SignInScheme = cookieScheme;
+
+                if (string.IsNullOrWhiteSpace(options.Authority))
+                    options.Authority = AuthorityHelpers.BuildAuthority(microsoftIdentityOptions);
+
+                if (!AuthorityHelpers.IsV2Authority(options.Authority))
+                    options.Authority += "/v2.0";
+                
+                options.TokenValidationParameters.NameClaimType = "preferred_username";
+
+                // If you want to restrict the users that can sign-in to several organizations
+                // Set the tenant value in the appsettings.json file to 'organizations', and add the
+                // issuers you want to accept to options.TokenValidationParameters.ValidIssuers collection
+                options.TokenValidationParameters.IssuerValidator = AadIssuerValidator.GetIssuerValidator(options.Authority).Validate;
+
```

Comment:
Provide option to override 
                if (options.TokenValidationParameters.IssuerValidator != null)
                {

Reply:
Fixed. All events are now taking in account custom code from the developer